lua/orgmode/objects/priority_state.lua

```diff
@@ -0,0 +1,98 @@
+local config = require('orgmode.config')
+local utils = require('orgmode.utils')
+
+---@class PriorityState
+local PriorityState = {}
+
+---@param current_priority string
+function PriorityState:new(current_priority)
+  local o = {}
+
+  o.current_priority = current_priority
+  o.high_priority = config.org_priority_highest
+  o.low_priority = config.org_priority_lowest
+
+  setmetatable(o, self)
+  self.__index = self
+
+  return o
+end
+
+---@return string
+function PriorityState:prompt_user()
+  local choice = utils.choose(self:_get_enumerated())
+  if not choice then
+    return
+  end
+
+  return choice.choice_value
+end
+
+function PriorityState:is_alphabetical()
+  return string.find(self.high_priority, '%a')
+end
+
+---@return string
+function PriorityState:increase()
+  if self.current_priority == self.high_priority then
+    print('increase return early')
+    return self.current_priority
+  end
+
+  return self:_apply(self:is_alphabetical() and -1 or 1)
```

Comment:
I just tested Emacs orgmode and it actually treats number priority the same way as char priority. Which means that `1` is highest priority, and `3` is the lowest. This means that you can always do `string.byte` and `string.char`, and it should give you the desired result.

Also, I figured out that my current implementation of getting the priority number for sorting is not correct. I always assumed that it's possible only to set one of those 3 config options, but it turns out you can also go in between, especially with numbers.

Could you update that method on section so it properly works with this change?

Reply:
I assume you are talking about [this](https://github.com/kristijanhusak/orgmode.nvim/blob/012a7eb5e292311d26301b882193add0bf4c058b/lua/orgmode/parser/section.lua#L193) section?

Reply:
Should be fixed, fixing tests then will push up.

Reply:
Yeah that section.

Reply:
Should be fixed, let me know how that looks.